### PR TITLE
[SECURITY] Secured Trunk and Glovebox

### PR DESCRIPTION
**Pull request by r0adra93**
Converted the the trunk and glovebox items events into functions and exports.

## Description

Replaced the glovebox and trunk events with functions and exports. Set it up in qb-jobs.
Needs to be merged along with qb-policejob: https://github.com/qbcore-framework/qb-policejob/pull/422
As the trunk script is insecure and it relies on the security update.

## Checklist

- [x] I have personally loaded this code into an updated qbcore project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

File: server/main.lua [hunk 1]
```diff
@@ -535,8 +535,6 @@ local function SetupShopItems(shopItems)
 end
 
 ---Get items in a stash
----@param stashId string The id of the stash to get
----@return table items
```
Comment (tom-osborne) [inline server/main.lua:539]: These are [luadoc](https://keplerproject.github.io/luadoc/manual.html#introduction) comments, please add these back in.

File: server/main.lua [hunk 2]
```diff
@@ -1126,20 +1122,20 @@ AddEventHandler('onResourceStart', function(resourceName)
 		end)
 	end
 end)
-
 RegisterNetEvent('QBCore:Server:UpdateObject', function()
     if source ~= '' then return end -- Safety check if the event was not called from the server.
     QBCore = exports['qb-core']:GetCoreObject()
 end)
-
-RegisterNetEvent('inventory:server:addTrunkItems', function(plate, items)
```
Comment (tom-osborne) [inline server/main.lua:1135]: If we are removing events, we should leave them in and just remove their functionality, replacing it with a helpful error message displayed in the console.

**Review by tom-osborne — CHANGES_REQUESTED** [review]

File: server/main.lua (hunk 2 above)
Comment (r0adra93) [reply server/main.lua:1135]: Only qb-police job uses it and I was making the change.

Comment (r0adra93) [reply server/main.lua:1135]: oh okay I can do that

File: server/main.lua [hunk 3]
```diff
@@ -567,8 +567,6 @@ local function GetStashItems(stashId)
 end
 
 ---Save the items in a stash
----@param stashId string The stash id to save the items from
```
Comment (charj) [inline server/main.lua:570]: seems a shame to remove perfectly good LDoc annotations

Comment (r0adra93) [reply server/main.lua]: Resolved

**Review by Irishstevie — APPROVED** [review]
I run this with qb-policejob: https://github.com/qbcore-framework/qb-policejob/pull/422

Tested storage Pull out cars open all the menus 

Works good

File: server/main.lua [hunk 4]
```diff
@@ -1059,6 +1059,220 @@ local function CreateNewDrop(source, fromSlot, toSlot, itemAmount)
 	end
 end
 
+local function OpenInventory(name, id, other, origin)
+	local src = origin
+	local ply = Player(src)
+    local Player = QBCore.Functions.GetPlayer(src)
+	if not ply.state.inv_busy then
```
Comment (charj) [inline server/main.lua:1066]: By flipping this condition you could bring your whole block of conditions down by one level of indentation
```
if not ply.state.inv_busy then 
    return QBCore.Functions.Notify(src, Lang:t("notify.noaccess"), 'error')
end
...
```

File: server/main.lua [hunk 5]
```diff
@@ -1059,6 +1059,220 @@ local function CreateNewDrop(source, fromSlot, toSlot, itemAmount)
...
+		if name and id then
```
Comment (charj) [inline server/main.lua:1067]: The same concept can be applied to this one also

Comment (r0adra93) [reply server/main.lua:1067]: You do realize when you comment; you need to approve the code.

**Review by Z3rio — APPROVED** [review]
The inventory seems to be working as normal after these changes.

**Comment by zwkf** [comment]
This PR being merged has created further issues:
https://github.com/qbcore-framework/qb-inventory/pull/422

**Comment by r0adra93** [comment]
> This PR being merged has created further issues: #422

Roll it back...

**Pull request merged**